### PR TITLE
Fix 9399

### LINKED ISSUE #9399: Incorrect query output from group by query (regression in 0.9.0)
### What happens?

Hi,
I have a query where I group by an integer column and I was expecting that I would only get unique values, however recent versions of duckdb (0.9.0+) return duplicates.

Query: 

```sql
with a as (select label,                                                        
   avg(x) as xx,                                                                
   avg(y) as yy                                                                 
from tab group by label)                                                        
select * from a 
```

Full reproduction is given below. 
The issue does not happen with duckdb 0.8.1

Thanks !
 

### To Reproduce

Here is the code

It requires a data-file  ( https://gist.github.com/segasai/36a73d6f3b140e513e1adfc5d05f2c83 )

```python
import duckdb
import numpy as np
import pandas as pd

pdf = pd.read_csv('aa.csv')
conn1 = duckdb.connect(':memory:')
conn1.register('tab', pdf)
R = conn1.execute('''                                                       
with a as (select label,
   avg(x) as xx,
   avg(y) as yy
from tab group by label)                            
select * from a 
                           
''').fetchnumpy()
print(len(R['label']), len(np.unique(R['label'])))
```
The code should print the length of the result and the number of unique values for the column used as a key. They must be equal but they are not. 

For clarity the underlying query is this 

```sql
with a as (select label,
   avg(x) as xx,
   avg(y) as yy
from tab group by label)                            
select * from a 
```

### OS:

Linux

### DuckDB Version:

0.9.1

### DuckDB Client:

Python

### Full Name:

Sergey Koposov

### Affiliation:

University of Edinburgh

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by lnkuiper**
Fixes #9399

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM. Is there also a test we could add here?

**Comment by lnkuiper** [comment]
Absolutely. I've written a test and verified that without the fix, the test fails

File: test/issues/general/test_9399.test [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# create table that's has just one row group, and just a few duplicates
+statement ok
+CREATE TABLE test AS
+SELECT CASE WHEN random() < 0.002 THEN range ELSE 100000 - range END c0,
+    random() c1,
+    random() c2
+FROM range(100000)
+
+mode output_hash
```
Comment (Mytherin) [inline test/issues/general/test_9399.test:21]: Can we remove the `mode output_hash`?

Comment (lnkuiper) [reply test/issues/general/test_9399.test]: Of course! I think this is ready to go now

**Pull request merged**